src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentSubmitterUTest.java

```diff
@@ -210,35 +266,67 @@ public void submitCourseAttendancesHappyPath() throws Exception {
         when(assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment1, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment1);
         when(assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment1, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment1);
-        when(assignmentValidator.validateConfigurationSetupExistence(attendanceSummaryModel1, attendanceAssignment1)).thenReturn(attendanceAssignment1);
+        when(assignmentValidator.validateConfigurationSetupExistence(attendanceAssignment1)).thenReturn(attendanceAssignment1);
         when(assignmentValidator.validateAttendanceAssignment(COURSE_ID, attendanceAssignment2, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment2);
         when(assignmentValidator.validateCanvasAssignment(assignmentConfigurationFromSetup, COURSE_ID, attendanceAssignment2, canvasApiWrapperService, oauthToken)).thenReturn(attendanceAssignment2);
-        when(assignmentValidator.validateConfigurationSetupExistence(attendanceSummaryModel2, attendanceAssignment2)).thenReturn(attendanceAssignment2);
+        when(assignmentValidator.validateConfigurationSetupExistence(attendanceAssignment2)).thenReturn(attendanceAssignment2);
+        when(sectionService.getFirstSectionOfCourse(course.getCanvasCourseId())).thenReturn(section1);
+        when(studentRepository.findByCanvasCourseId(course.getCanvasCourseId())).thenReturn(studentList);
+        when(studentService.getStudent(STUDENT_1_ID)).thenReturn(student1);
+        when(studentService.getStudent(STUDENT_2_ID)).thenReturn(student2);
+        when(studentService.getStudent(STUDENT_3_ID)).thenReturn(student3);
+        when(studentService.getStudent(STUDENT_4_ID)).thenReturn(student4);
+        when(studentService.getStudentByCourseAndSisId(SIS_USER_ID_1, course.getCanvasCourseId())).thenReturn(student1List);
+        when(studentService.getStudentByCourseAndSisId(SIS_USER_ID_2, course.getCanvasCourseId())).thenReturn(student2List);
+        when(studentService.getStudentByCourseAndSisId(SIS_USER_ID_3, course.getCanvasCourseId())).thenReturn(student3List);
+        when(studentService.getStudentByCourseAndSisId(SIS_USER_ID_4, course.getCanvasCourseId())).thenReturn(student4List);
+        when(sectionService.getSection(SECTION_1_ID)).thenReturn(section1);
+        when(sectionService.getSection(SECTION_2_ID)).thenReturn(section2);
 
         assignmentSubmitter.submitCourseAttendances(true, attendanceSummaryModelList, COURSE_ID, oauthToken, assignmentConfigurationFromSetup);
-        verify(canvasApiWrapperService, times(2)).gradeMultipleSubmissionsBySection(any(), any());
+        verify(canvasApiWrapperService, times(1)).gradeMultipleSubmissionsByCourse(any(), any());
     }
 
     @Test
     public void submitCourseAttendancesFailCanvasGradingError() throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS00112)